src/config.ts

```diff
@@ -39,22 +39,28 @@ export function defineConfig(config: ViteUserConfig = {}): ViteUserConfig {
         },
       },
       forceRerunTriggers: [
-        // Vitest defaults
-        '**/package.json/**',
-        '**/{vitest,vite}.config.*/**',
+        ...(_config.test?.forceRerunTriggers
+          ? []
+          : [
+              // Vitest defaults
+              '**/package.json/**',
+              '**/{vitest,vite}.config.*/**',
+            ]),
         // Re-run tests when Nitro is rebuilt
         join(
-          _config.nitro?.rootDir || '',
+          nitro?.rootDir || '',
           NITRO_OUTPUT_DIR,
-          _config.nitro?.mode === 'production' ? 'server' : '.nitro/dev',
+          nitro?.mode === 'production' ? 'server' : '.nitro/dev',
           'index.mjs',
         ),
       ],
       globalSetup: [
         join(currentDir, 'setup.mjs'),
       ],
       // @ts-expect-error: Append Nitro config to access in global setup file
-      nitro: _config.nitro,
+      nitro,
     },
-  })
+  }) as ViteUserConfig
```

Comment:
**Note:** Had to add this type assertion because the `// @ts-expect-error` changes the return type of `defineVitestConfig()` from:

```ts
UserConfig
```

to:

```ts
UserConfig & Promise<UserConfig> & (UserConfigFnObject & UserConfigExport)
```

Which is [disallowed] by `mergeConfig()`:

```
export function mergeConfig<
  D extends Record<string, any>,
  O extends Record<string, any>,
>(
  defaults: D extends Function ? never : D,
  overrides: O extends Function ? never : O,
  isRoot = true,
): Record<string, any> {
  // ...
}
```

[disallowed]: https://github.com/vitejs/vite/blob/v5.3.3/packages/vite/src/node/utils.ts#L1130-L1137

Reply:
Makes sense!